clients/src/main/java/org/apache/kafka/common/ConfigEntry.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common;
+
+public class ConfigEntry {
+    private String configKey;
+    private String configValue;
+
+    public ConfigEntry(String configKey, String configValue) {
+        this.configKey = configKey;
```

Comment:
Though "not null" requirement makes sense, ConfigEntry is just a DTO, we usually don't add business logic to default constructors (like in PartitionState etc).